NMA-301 Shortcut Button | Create Username

tomasz-ludek opened the pull request.

HashEngineering commented on wallet/src/de/schildbach/wallet/service/BlockchainServiceImpl.java:
```diff
@@ -763,6 +764,8 @@ public void onCreate() {
         application.getWallet().addChangeEventListener(Threading.SAME_THREAD, walletEventListener);
 
+        application.getWallet().getContext().sporkManager.addEventListener(sporkUpdatedEventListener, Threading.SAME_THREAD);
+
```
The `sporkManager.addEventlistener` and related code was removed because this line in particular was causing crashes in some cases when the app was installed and run on new devices through the Google Firebase or the Prelaunch Report mechanism.

Since sporks were not used for anything in the app, I removed this code on the master branch at https://github.com/dashevo/dash-wallet/commit/e237a37120d454c507614dab46c8a7055262997c (#334)

tomasz-ludek commented:
@HashEngineering ,
I added the sporkUpdatedEventListener related code back as a reference for implementing the requirement of hiding the "Join DashPay" action if platform-related sporks are disabled. I liked my old implementation of handling SPORK_16_INSTANTSEND_AUTOLOCKS state changing and wanted to apply something similar over here.
Anyways, as it is not supported anymore, we will find another solution when it's time.

I removed all the code related to sporkUpdatedEventListener.

tomasz-ludek commented:
Sorry, I closed this PR by accident. Opening back.

HashEngineering commented:
> I added the sporkUpdatedEventListener related code back as a reference for implementing the requirement of hiding the "Join DashPay" action if platform-related sporks are disabled. I liked my old implementation of handling SPORK_16_INSTANTSEND_AUTOLOCKS state changing and wanted to apply something similar over here.
> Anyways, as it is not supported anymore, we will find another solution when it's time.
> 
> I removed all the code related to sporkUpdatedEventListener.

So far it looks like there are no Platform related sporks, since Platform is currently separate from the Dash Core.

sambarboza commented on wallet/src/de/schildbach/wallet/ui/WalletActivity.java:
```diff
@@ -1036,11 +1064,14 @@ public void onEvent(SyncProgressEvent event) {
                 syncStatusTitle.setText(R.string.sync_status_sync_title);
                 syncStatusMessage.setText(R.string.sync_status_sync_completed);
                 showSyncPane(R.id.sync_status_pane, false);
+                syncComplete = true;
```
Can we remove this line?

tomasz-ludek replied:
Yeah, it doesn't make sense ;P
Strange that Lint didn't warn me.
Removed, thanks!

tomasz-ludek commented:
@sambarboza issue fixed.
